src/menu_bar.rs

```diff
@@ -0,0 +1,106 @@
+//! # TreeView Sample
+//!
+//! This sample demonstrates how to create a TreeView with either a ListStore or TreeStore.
+
+extern crate gtk;
+
+use gtk::{
+    AboutDialog, BoxExt, CheckMenuItem, ContainerExt, IconSize, Image, Inhibit, Label,
+    Menu, MenuBar, MenuItem, MenuItemExt, MenuShellExt, WidgetExt, WidgetSignals,
+    Window, WindowExt, WindowPosition, WindowType
+};
+
+fn main() {
+    if gtk::init().is_err() {
+        println!("Failed to initialize GTK.");
+        return;
+    }
+
+    let window = Window::new(WindowType::Toplevel);
+
+    window.set_title("MenuBar example");
+    window.set_position(WindowPosition::Center);
+    window.set_size_request(400, 400);
+
+    window.connect_delete_event(|_, _| {
+        gtk::main_quit();
+        Inhibit(false)
+    });
+
+    let h_box = gtk::Box::new(gtk::Orientation::Vertical, 10);
+
+    let menu = Menu::new();
+    let menu_bar = MenuBar::new();
+    let file = MenuItem::new_with_label("File");
+    let about = MenuItem::new_with_label("About");
+    let quit = MenuItem::new_with_label("Quit");
+    let file_entry = MenuItem::new();
+    let file_box = gtk::Box::new(gtk::Orientation::Horizontal, 0);
+    let file_image = Image::new_from_file("resources/file.png");
+    let file_label = Label::new(Some("File"));
+    let folder_entry = MenuItem::new();
+    let folder_box = gtk::Box::new(gtk::Orientation::Horizontal, 0);
+    let folder_image = Image::new_from_icon_name("folder-music-symbolic", IconSize::Menu.into());
+    let folder_label = Label::new(Some("Folder"));
+    let check_entry = CheckMenuItem::new_with_label("Click me!");
+
+    file_box.pack_start(&file_image, false, false, 0);
+    file_box.pack_start(&file_label, true, true, 0);
+    file_entry.add(&file_box);
+    folder_box.pack_start(&folder_image, false, false, 0);
+    folder_box.pack_start(&folder_label, true, true, 0);
+    folder_entry.add(&folder_box);
+    menu.append(&file_entry);
+    menu.append(&folder_entry);
+    menu.append(&check_entry);
+    menu.append(&about);
+    menu.append(&quit);
+    file.set_submenu(Some(&menu));
+    menu_bar.append(&file);
```

Comment:
There's a chance we're unaware of some obstacles to using Glade, especially around menus and actions. The GTK learning materials push strongly toward using Glade, doing a lot of these manipulations manually seems unidiomatic. So we need to prioritize making sure the Glade way works.